docs/best-practices.md

```diff
@@ -0,0 +1,41 @@
+### How to select good environmental variable name?
+When selecting for environmental variable name, please check Dotnet cli convention examples.
+Please check discussion [here](https://github.com/NuGet/Client.Engineering/issues/882#issuecomment-816118683) for more details, please note this's not ultimate guide just our learning from last experience and hoping to help you, feel free to improve it. 
+Since env var name is case sensitive in Linux, so please use all uppercase (separated by underscore) naming convention for best practice.  But value doesn't have to case sensitive, we can convert to [uppercase](https://github.com/NuGet/NuGet.Client/blob/680f9bd4e97db7cd7482584276886764de69d3cb/src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs#L530) and compare to validate value.
+We selected in DOTNET_NUGET_SIGNATURE_VERIFICATION for 'Add an environment variable to opt-in to the package signing verification on .NET 5+'. Let's dissect part of env var name.
+
+<b>DOTNET_NUGET_SIGNATURE_VERIFICATION</b>
+
+When selecting name first/second parts indicate where this env variable applies. In this case it applies to <b>DOTNET</b> and more specifically for <b>NUGET</b> use cases. 
```

Comment:
I think you've done a great job of disecting the name. I think you can make it clear by saying the env var should be something like: `PRODUCT_FEATURE_OPERATION`

Reply:
Addressed. Please check now.